src/lib.rs

```diff
@@ -345,12 +346,27 @@ pub mod pallet {
 
 			let current_initialized_rewards = InitializedRewardAmount::<T>::get();
 
-			// This ensures that the rewards match the pot
+			let reward_difference = Self::pot().saturating_sub(current_initialized_rewards);
+
+			// Ensure the difference is not bigger than the total number of contributors
 			ensure!(
-				current_initialized_rewards == Self::pot(),
+				reward_difference < TotalContributors::<T>::get().into(),
 				Error::<T>::RewardsDoNotMatchFund
 			);
 
+			if reward_difference > 0u32.into() {
```

Comment:
Although it's harmless, I don't see why it's necessary to make this `> 0` comparison. If the difference is zero then the burn is a noop anyway.